internal/collector/lidarr/music.go

```diff
@@ -0,0 +1,219 @@
+		),
+		songsMonitoredMetric: prometheus.NewDesc(
+			"lidarr_songs_monitored_total",
+			"Total number of monitored songs",
+			nil,
+			prometheus.Labels{"url": c.String("url")},
+		),
+		songsDownloadedMetric: prometheus.NewDesc(
+			"lidarr_songs_downloaded_total",
+			"Total number of downloaded songs",
+			nil,
+			prometheus.Labels{"url": c.String("url")},
+		),
+		songsMissingMetric: prometheus.NewDesc(
+			"lidarr_songs_missing_total",
+			"Total number of missing songs",
+			nil,
+			prometheus.Labels{"url": c.String("url")},
+		),
+		songsQualitiesMetric: prometheus.NewDesc(
+			"lidarr_songs_quality_total",
+			"Total number of downloaded songs by quality",
+			[]string{"quality"},
+			prometheus.Labels{"url": c.String("url")},
+		),
+	}
+}
+
+func (collector *lidarrCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- collector.artistsMetric
+	ch <- collector.artistsMonitoredMetric
+	ch <- collector.artistGenresMetric
+	ch <- collector.artistsFileSizeMetric
+	ch <- collector.albumsMetric
+	ch <- collector.albumsMonitoredMetric
+	ch <- collector.albumsGenresMetric
+	ch <- collector.songsMetric
+	ch <- collector.songsMonitoredMetric
+	ch <- collector.songsDownloadedMetric
+	ch <- collector.songsMissingMetric
+	ch <- collector.songsQualitiesMetric
+}
+
+func (collector *lidarrCollector) Collect(ch chan<- prometheus.Metric) {
+	c := client.NewClient(collector.config)
+
+	var artistsFileSize int64
+	var (
+		artistsMonitored = 0
+		artistGenres     = map[string]int{}
+		albums           = 0
+		albumsMonitored  = 0
+		albumGenres      = map[string]int{}
+		songs            = 0
+		songsDownloaded  = 0
+		songsQualities   = map[string]int{}
+	)
+
+	artists := model.Artist{}
+	if err := c.DoRequest("artist", &artists); err != nil {
+		log.Fatal(err)
+	}
+
+	for _, s := range artists {
+		if s.Monitored {
+			artistsMonitored++
+		}
+		albums += s.Statistics.AlbumCount
+		songs += s.Statistics.TotalTrackCount
+		songsDownloaded += s.Statistics.TrackFileCount
+		artistsFileSize += s.Statistics.SizeOnDisk
+
+		for _, genre := range s.Genres {
+			artistGenres[genre]++
+		}
+
+		if collector.config.Bool("enable-song-quality-metrics") {
+			songFile := model.SongFile{}
+			if err := c.DoRequest(fmt.Sprintf("%s?artistid=%d", "trackfile", s.Id), &songFile); err != nil {
+				log.Fatal(err)
+			}
+			for _, e := range songFile {
+				if e.Quality.Quality.Name != "" {
+					songsQualities[e.Quality.Quality.Name]++
+				}
+			}
+		}
+
+		if collector.config.Bool("enable-monitored-albums-metrics") {
+			album := model.Album{}
+			if err := c.DoRequest(fmt.Sprintf("%s?artistid=%d", "album", s.Id), &album); err != nil {
+				log.Fatal(err)
+			}
+			for _, a := range album {
+				if a.Monitored {
+					albumsMonitored++
+				}
+				for _, genre := range s.Genres {
+					albumGenres[genre]++
+				}
+			}
+		}
+	}
+
+	songMissing := model.Missing{}
+	if err := c.DoRequest("wanted/missing?sortKey=airDateUtc", &songMissing); err != nil {
```

Comment:
```suggestion
	if err := c.DoRequest("wanted/missing", &songMissing); err != nil {
```

Having that filter causes a HTTP 400 error

Reply:
Weird, I'm able to sort by that key and `releaseDate`, which makes more sense. Can you test?

Reply:
Does it work without the `sortKey` for you?  We shouldn't need it if it does. With radarr and sonarr there was a bug that required a `sortKey` that's why it's included in those.

Reply:
You're right, no need to sort tho!

Reply:
This may have been fixed in the Sonarr and Radarr API too, but I haven't had time to test.